musicbrainz/csv/convert_to_csv.py

```diff
@@ -1,69 +1,134 @@
-                value[k + "_id"] = v["id"]
+    if "relations" in key or "aliases" in key or "tags" in key:
+        # ignore relations, aliases and tags to make output simplier
+        return
+
+    if isinstance(data, dict):
+        # the input JSON Lines format is lines of dictionaries, and the input data should be
+        # a list of dictionaries.
+        if first_level:
+            # if this dictionary is not nested i.e. first level, then we extract every entry.
+            global values
+            # make a new entry for the value list, since each first level dictionary
+            # in JSON file represents a new instance. This value dictionary is carried and 
+            # preserved between each recursive call.
+            value = {}
+            for k in data:
+                extract(data[k], value, False, k)
             
-            if "name" in v:
-                if k + "_name" not in header:
-                    header.append(k + "_name")
-                value[k + "_name"] = v["name"]
-                            
-    values.append(value)
+            # after extracting every entry of the current line, append it to the list and empty it.
+            values.append(copy.deepcopy(value))
+            value = {}
+
+        else:
+            # if this dictionary is nested, then we do not extract all info,
+            # we only need the id and the name of that dictionary.
+            for k in data:
+                if k == "id":
+                    # extract its id
+                    extract(data["id"], value, first_level, key + "_id")
+
+                if k == "name":
+                    # extract its name
+                    extract(data["name"], value, first_level, key + "_name")
+
+                if isinstance(data[k], dict) or isinstance(data[k], list):
+                    # if there is still a nested instance, extract further
+                    extract(data[k], value, first_level, key + "_" + k)
+
+    elif isinstance(data, list):
+        # extract each element of the list.
+        for element in data:
+            extract(element, value, first_level, key)
                 
-print(header)
-# print(values)
+    else:
+        # if data is not a collection, we parse it and add to the current value dictionary.
+        global header
+        if key not in header:
+            header.append(key)
 
-with open("out.csv", "w") as out:
-    # write header
-    line = "id,"
-    for column in header:
-        if column == "id":
-            continue
-            
-        line += column
+        v = data
+        if isinstance(data, str):
+            v = v.replace(",", "\,")
+            v = v.replace("\r\n", "")
+
+        if data is None:
+            v = ""
+
+        try: 
+            value[key]
+        except KeyError:
+            value[key] = v
+        else:
+            if isinstance(value[key], list): (value[key]).append(v)
+            else: value[key] = [value[key]] + [v]
```

Comment:
This is really hard to debug, and it's absolutely bonkers to me that that you would raise an exception just to get the functionality of an `if` statement, but I'm pretty sure it can be rewritten something like this:

```suggestion
    if key in value:
            if isinstance(value[key], list): 
                value[key].append(v)
            else: 
                value[key] = [value[key]] + [v]
    else:
        value[key] = v
```

I won't comment on how the code is actually functioning...